Docker.DotNet.BasicAuth/MaybeSecureString.cs

```diff
@@ -0,0 +1,74 @@
+using System;
+using System.Runtime.InteropServices;
+using System.Security;
+
+namespace Docker.DotNet.BasicAuth
+{
+    internal struct MaybeSecureString : IDisposable
+    {
+#if !netstandard
+        public SecureString Value;
+
+        public MaybeSecureString(string str)
+        {
+            var secureStr = new SecureString();
+
+            if (str.Length > 0)
+            {
+                foreach (char c in str)
+                {
+                    secureStr.AppendChar(c);
+                }
+            }
+
+            Value = secureStr;
+        }
+
+        public MaybeSecureString(SecureString value)
+        {
+            Value = value;
+        }
+
+        public void Dispose()
+        {
+            Value.Dispose();
```

Comment:
These seem to conflict. Value.Dispose() on null would throw but .Empty compares to null? How can we be in this case ever?